### PR TITLE
TEC-7957 - Checkstyle (update checkstyle.xml for LinkedIn project)

### PR DESCRIPTION
**Pull request by RadostinEcho**
### Spec
N/A

### Priority
Minor

### Description of Changes

- Add comments in:
        main.echobox.api.linkedin.client
        linkedin.connection
        linkedin.exception 
        linkedin.types.assets
        linkedin.types
        linkedin.jsonmapper

- Edit checksyle.xml file in LinkedIn project to check for package-private methods and fields.

### Documentation
N/A

### Risks & Impacts
No potential risks

### Security
N/A

### Testing
Ensured project compiled successfully by checking with checkstyle.xml.
```
<module name="JavadocMethod">
        <property name="scope" value="package"/>
    <property name="allowedAnnotations" value="Override, Test, Mock, Before"/>
</module>
<module name="JavadocVariable">
    <property name="scope" value="package"/>
</module>
```
Checked manually that the classes don't fail the test.
### Deployment Considerations
N/A

### CODE REVIEW

File: src/main/java/com/echobox/api/linkedin/client/Parameter.java [hunk 1]
```diff
@@ -85,7 +85,7 @@ private Parameter(String name, Object value, JsonMapper jsonMapper) {
    * <p>
    * The {@code value} parameter is often a {@link String} or primitive type like {@link Integer}, 
    * but you may pass in a {@link java.util.List}, {@link java.util.Map}, or your own 
-   * <tt>@LinkedIn</tt>-annotated Javabean and it will be converted to JSON automatically. 
+   * {@code LinkedIn-annotated} Javabean and it will be converted to JSON automatically.
    *
```
Comment (RadostinEcho) [inline src/main/java/com/echobox/api/linkedin/client/Parameter.java]: {@code LinkedIn-annotated} or {@code LinkedIn}-annotated

File: src/main/java/com/echobox/api/linkedin/client/Parameter.java [hunk 2]
```diff
@@ -105,7 +105,7 @@ public static Parameter with(String name, Object value) {
    * <p>
    * The {@code value} parameter is often a {@link String} or primitive type like {@link Integer},
    * but you may pass in a {@link java.util.List}, {@link java.util.Map}, or your own
-   * <tt>@LinkedIn</tt>-annotated Javabean and it will be converted to JSON automatically. 
+   * {@code LinkedIn-annotated} Javabean and it will be converted to JSON automatically.
    *
```
Comment (RadostinEcho) [inline src/main/java/com/echobox/api/linkedin/client/Parameter.java]: {@code LinkedIn-annotated} or {@code LinkedIn}-annotated

File: src/main/java/com/echobox/api/linkedin/client/Parameter.java (hunk 1 above)
Comment (eddspencer) [reply src/main/java/com/echobox/api/linkedin/client/Parameter.java]: How about `{@link LinkedIn} annotated`?

Comment (RadostinEcho) [reply src/main/java/com/echobox/api/linkedin/client/Parameter.java]: Sure, it seems that is how the original one should have looked like.

**Pull request merged**